rpcs3/Emu/NP/upnp_handler.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <condition_variable>
+#include <unordered_map>
+
+#include <miniupnpc.h>
+
+#include "upnp_config.h"
+#include "Utilities/mutex.h"
+
+class upnp_handler
+{
+public:
+	~upnp_handler();
+
+	void upnp_enable();
+	void add_port_redir(std::string_view addr, u16 internal_port, std::string_view protocol);
+	void remove_port_redir(u16 internal_port, std::string_view protocol);
+
+	bool is_active() const;
+
+private:
+	void remove_port_redir_external(u16 external_port, std::string_view protocol);
+
+private:
```

Comment:
duplicate

Reply:
Code style choice to separate variables from functions/